gotham/src/router/route/mod.rs

```diff
@@ -49,6 +49,7 @@ pub enum Delegation {
 /// `Route` exists as a trait to allow abstraction over the generic types in `RouteImpl`. This
 /// trait should not be implemented outside of Gotham.
 pub trait Route: RefUnwindSafe {
+    type ResBody;
```

Comment:
Nitpicky, but this is probably better as `ResponseBody` just because it's a type. Or maybe just `BodyType` or something. `Body` would be ideal.

I know it's in sync with Hyper, but it might be nice to distance ourselves a little bit here.